posthog-core/src/index.ts

```diff
@@ -1593,13 +1596,15 @@ export abstract class PostHogCore extends PostHogCoreStateless {
     return this._decideAsync(sendAnonDistinctId)
   }
 
-  private cacheSessionReplay(response?: PostHogRemoteConfig): void {
+  private cacheSessionReplay(source: string, response?: PostHogRemoteConfig): void {
     const sessionReplay = response?.sessionRecording
     if (sessionReplay) {
       this.setPersistedProperty(PostHogPersistedProperty.SessionReplay, sessionReplay)
-      this.logMsgIfDebug(() => console.log('PostHog Debug', 'Session replay config: ', JSON.stringify(sessionReplay)))
-    } else {
-      this.logMsgIfDebug(() => console.info('PostHog Debug', 'Session replay config disabled.'))
+      this.logMsgIfDebug(() => console.log('PostHog Debug', `Session replay config from ${source}: `, JSON.stringify(sessionReplay)))
+    } else if (typeof sessionReplay === 'boolean' && sessionReplay === false) {
```

Comment:
we only clear the data if sessionReplay=false because /flags does not return `sessionRecording` yet otherwise its gonna overwrite the cached value from the remote config
/decide?=v4 and remote config do return the sessionRecording config

Reply:
🙌